static/bidder-info/iionads.yaml

```diff
@@ -0,0 +1,16 @@
+endpoint: "https://{{.Host}}/openrtb/{{.PublisherID}}"
```

Comment:
This is so weird to see URL like this, but I guess it's possible too. 
Do you want to add validation for the "host" parameter to make sure it's URL, not just a random set of characters?

Reply:
Thank you. We discussed this with the team. Unfortunately this URL format may lead to a potential security issue. We recommend to modify it to something like this (based on unit tests you provided):
"https://www.limelightDigital.{{.MediaType}}.test.com/openrtb/{{.PublisherID}}"

Reply:
Correct @VeronikaSolovei9. We do not allow the domain name to be configurable due to security issues. Additionally, we highly discourage subdomains as they negatively affect the performance of maintaining connects to your bidding server. Please see our bidder development guide for further information: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html

I am aware there are a few adapters currently in violation. The Prebid Server Committee is working on a remediation plan.

Reply:
As noted in the docs PR https://github.com/prebid/prebid.github.io/pull/4297, we need to support the same parameters with our Prebid.js adapter. Our Prebid.js has `Host` parameter. And a lot of our client use it on their custom domains which are owned by our clients and which we do not directly control. Are you suggesting we set up cname dns records from our global domain to their custom domains and hope to win performance?

Reply:
Is it perfect solution if we will create a new alias adapter for each our client with his own domain? No subdomains. Ho `Host` parameters. Will it be easier to maintain? Will you win in performance and connections?

Reply:
Hi @imedvedko. Thanks for your patience and for working with us on this feedback.

The two main concerns we have for using a macro in the domain are security exposure and connection performance. The security aspect is alleviated by using a fixed top level domain. Due to the potential harm to hosts, we are strict in this requirement. We are working towards fixing  the few adapters currently in violation.

The connection performance advice is for your benefit. Client specific subdomains prevent Prebid Server from reusing connections across your clients which results in more connections needed for your adapter. The issue gets worse the more successful you are in attracting new clients. Hosts may choose to disable your adapter due to this behavior.

The best practice is for your domain to be fixed and any parameters specified in the path or query. How you choose to forward bid requests to your clients is outside of the scope of PBS and for your team to decide. I am hesitant to advise on this but my gut is that cnames are not the answer but perhaps it could be accomplished with vhosts, url rewrites or some other means at the edge.

Please do not create an alias for each client. That is not the intention of this feature.

I hope that helps.

Reply:
Also note that only the PBS host can see the URL used to connect to your bidder. Well, a debug auction can reveal the http call(s) sent out to the bidder, but beyond that it is completely invisible. So there is little vanity reason for having multiple URLs. A structural reason may exist in the event that each URL goes to a different physical host which each have a different dataset loaded to represent each client. But as Brian points out above, this doesn't seem like a very scalable architecture.

Reply:
Hi, sorry for the long answer. I added top level domain. For compatibility with our Prebid.js adapter parameters, I left `host` field format in the request the same and just cut out subdomain from it. I also added additional validation of `host` parameter to improve the security of subdomain cutting out https://github.com/prebid/prebid-server/pull/2539/commits/e0b1ab238714f95dea2150f81d3ca3ab4f65b34b#diff-85e257afc296b1b9772d361fe8269082ee7a158207ed84f6fae16e1b42a2426cR12 and added additional params tests https://github.com/prebid/prebid-server/pull/2539/commits/e0b1ab238714f95dea2150f81d3ca3ab4f65b34b#diff-98d261a63ffeb65e201644a1669caaa0d9e6c2fb2a9a9e6de7d429d22814b017R61-R70 . Thanks

Reply:
I have improved parameters validation through json schema in previous commits. I also added additional domain validation for host parameter in the adapter code in https://github.com/prebid/prebid-server/pull/2539/commits/7b2cc3abd0eb757108d22990267240fa63ca9351 commit. Feel free to request me to rollback this commit if these changes seem redundant to you

Reply:
This should prevent PBS from being used as a malicious proxy, unless a PBS host or future PR does something silly like define the endpoint as "https://{{.HOST}}.com/openrtb/..." It still has the potential of cluttering up the connection pool depending on how many unique HOST values the adapter might see as it runs on a host. But at least that is just a performance issue rather than a security issue.